### PR TITLE
Fix/adaln remove silu

### PR DESCRIPTION
**Pull request by lambertwjh**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by lambertwjh** [comment]
This is a small CogView4 alignment fix：
What this PR does

CogView4: remove the SiLU right before the final AdaLN.
Add a small switch use_silu (default True) to AdaLayerNormContinuous.
In CogView4Transformer2DModel, set norm_out = AdaLayerNormContinuous(..., use_silu=False) and keep SiLU only for the block path (temb_blocks = F.silu(temb_raw)) while feeding the raw temb_raw to the final AdaLN.

Why

In Megatron (our reference), adaln_final is a plain Linear (no activation).
In diffusers, temb was passed through F.silu before norm_out, which makes the final modulation differ from Megatron and causes hidden-state   @@mismatches.
This PR aligns diffusers’ CogView4 with Megatron’s behavior.

**Comment by zRzRzRzRzRzRzR** [comment]
@a-r-r-o-w can you check this

**Comment by a-r-r-o-w** [comment]
Instead of making this change here, can we just copy the layer into the CogView transformer files where required and use the correct code?

**Comment by lambertwjh** [comment]
“CogView4-only fix: final AdaLN without SiLU to match Megatron; generic normalization remains unchanged.”
please check again @a-r-r-o-w

File: src/diffusers/models/transformers/transformer_cogview4.py [hunk 1]
```diff
@@ -584,6 +583,37 @@ def forward(self, hidden_states: torch.Tensor) -> Tuple[torch.Tensor, torch.Tens
         return (freqs.cos(), freqs.sin())
 
 
+class _CogViewFinalAdaLayerNormContinuous(nn.Module):
```
Comment (a-r-r-o-w) [inline src/diffusers/models/transformers/transformer_cogview4.py]:
```suggestion
class CogView4AdaLayerNormContinuous(nn.Module):
```

File: src/diffusers/models/transformers/transformer_cogview4.py [hunk 2]
```diff
@@ -666,7 +696,7 @@ def __init__(
         )
 
         # 4. Output projection
-        self.norm_out = AdaLayerNormContinuous(inner_dim, time_embed_dim, elementwise_affine=False)
+        self.norm_out = _CogViewFinalAdaLayerNormContinuous(inner_dim, time_embed_dim, elementwise_affine=False)
```
Comment (a-r-r-o-w) [inline src/diffusers/models/transformers/transformer_cogview4.py]:
```suggestion
        self.norm_out = CogView4AdaLayerNormContinuous(inner_dim, time_embed_dim, elementwise_affine=False)
```

**Comment by a-r-r-o-w** [comment]
@bot style

**Comment by a-r-r-o-w** [comment]
@bot /style

**Comment by a-r-r-o-w** [comment]
@lambertwjh @zRzRzRzRzRzRzR Just to confirm, do the changes for the layer name look good to you? If so, I'll merge this :)

**Comment by lambertwjh** [comment]
Looks good to me. The rename to CogView4AdaLayerNormContinuous is clearer and matches the CogView4-only usage. Thanks! @a-r-r-o-w
